sdk/ai/Azure.AI.Agents.Persistent/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -1222,25 +1229,63 @@ public static IReadOnlyDictionary<string, string> GetToolCallAttributes(RunStepT
+            return toolCall switch
+            {
+                RunStepAzureAISearchToolCall azureSearch => azureSearch.AzureAISearch,
+                RunStepBingCustomSearchToolCall bingCustomSearch => bingCustomSearch.BingCustomSearch,
+                RunStepBingGroundingToolCall bingGrounding => bingGrounding.BingGrounding,
+                RunStepMicrosoftFabricToolCall fabric => fabric.MicrosoftFabric,
+                RunStepOpenAPIToolCall openApi => openApi.OpenAPI,
+                RunStepSharepointToolCall sharepoint => sharepoint.SharePoint,
+                _ => null // Return null for tool calls that don't have dictionary attributes (e.g. RunStepFunctionToolCall, RunStepCodeInterpreterToolCall)
+            };
+        }
+    }
+
+    internal class InstructionsEvent
+    {
+        [JsonPropertyName("content")]
+        public string Content { get; set; }
+    }
 
-            // Create the attribute name by removing "RunStep" and "ToolCall" from the class name
-            string attributeName = actualType.Name
-                .Replace("RunStep", string.Empty)
-                .Replace("ToolCall", string.Empty);
+    internal class MessageEvent
+    {
+        [JsonPropertyName("content")]
+        public string Content { get; set; }
+        [JsonPropertyName("role")]
+        public string Role { get; set; }
+    }
 
-            // Try to find a property with the parsed name
-            PropertyInfo property = actualType.GetProperty(attributeName, BindingFlags.Public | BindingFlags.Instance);
+    internal class RoleEvent
+    {
+        [JsonPropertyName("role")]
+        public string Role { get; set; }
+    }
 
-            if (property != null && property.PropertyType == typeof(IReadOnlyDictionary<string, string>))
-            {
-                // Get the value of the property
-                return property.GetValue(toolCall) as IReadOnlyDictionary<string, string>;
-            }
+    internal class ToolOutputEvent
+    {
+        [JsonPropertyName("content")]
+        public string Content { get; set; }
+        [JsonPropertyName("id")]
+        public string Id { get; set; }
+    }
 
-            // Return null if the property is not found or is not of the expected type
-            return null;
-        }
+    internal class ToolCallsEvent
+    {
+        [JsonPropertyName("tool_calls")]
+        public List<Dictionary<string, object>> ToolCalls { get; set; }
+    }
+
+    [JsonSerializable(typeof(InstructionsEvent))]
+    [JsonSerializable(typeof(MessageEvent))]
+    [JsonSerializable(typeof(RoleEvent))]
+    [JsonSerializable(typeof(ToolOutputEvent))]
+    [JsonSerializable(typeof(ToolCallsEvent))]
+    [JsonSerializable(typeof(string))]
+    [JsonSerializable(typeof(JsonElement))]
+    [JsonSerializable(typeof(Dictionary<string, object>))]
+    [JsonSerializable(typeof(List<Dictionary<string, object>>))]
+    internal partial class OpenTelemetryJsonContext : JsonSerializerContext
```

Comment:
Any reason this isn't ModelReaderWriterContext to match all other projects in azure-sdk-for-net?